src/main/java/dev/devature/penguin_api/service/IssueService.java

```diff
@@ -0,0 +1,127 @@
+package dev.devature.penguin_api.service;
+
+import dev.devature.penguin_api.entity.Issue;
+import dev.devature.penguin_api.enums.IssueResult;
+import dev.devature.penguin_api.repository.IssueRepository;
+import dev.devature.penguin_api.repository.OrganizationRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Optional;
+
+@Service
+@Transactional
+public class IssueService {
+    private IssueRepository issueRepository;
+    private OrganizationRepository organizationRepository;
+
+    @Autowired
+    public IssueService(IssueRepository issueRepository, OrganizationRepository organizationRepository){
+        this.issueRepository = issueRepository;
+        this.organizationRepository = organizationRepository;
+    }
+
+    /**
+     * @param issue Takes in an {@code Issue} that will be created in the database.
+     * @return The result of the request to the service layer as a {@code IssueResult}.
+     */
+    public IssueResult createIssue(Issue issue){
+        if(checkIfBadData(issue)){
```

Comment:
One major flaw with this flow is that it assumes the issue's `createdByID` is supplied by the API caller, and that the API caller isn't lying to us about who they are. As it stands, this flow is susceptible to a bug where user Alice calls the API to create a new issue, but supplies user Bob's user ID instead. So the Issue is created by Alice, but Bob is listed as the creator. Instead, we should have the API automatically set the calling user as the creator. We can get this information from the auth token, since this will uniquely identify the user who is calling our API. Check the flow I used for the Organizations endpoint. [This](https://github.com/Devature/penguin-api/blob/74e021165e31495cdd5bc08497607021ecf1ed80/src/main/java/dev/devature/penguin_api/service/OrganizationService.java#L52) is how I got the calling user's email address, and [this](https://github.com/Devature/penguin-api/blob/74e021165e31495cdd5bc08497607021ecf1ed80/src/main/java/dev/devature/penguin_api/controller/OrganizationController.java#L39) is how to get the claims from the auth token.